src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -127,13 +173,451 @@ int main(int argc, char* argv[])
 
 }
 
-
 void     showClInfo()
 {
     cout << "MUSI6106 Assignment Executable" << endl;
     cout << "(c) 2014-2022 by Alexander Lerch" << endl;
-    cout  << endl;
+    cout << endl;
 
     return;
 }
 
+bool test1();
+void test2();
+bool test3();
+bool test4();
+bool test5();
+
+void runTests()
+{
+    bool test1passed = test1();
+    test2();
+    bool test3passed = test3();
+    bool test4passed = test4();
+    bool test5passed = test5();
+}
+
+//Display Input and Output Audio Buffers
+void displayIOBuffers(float** ppfAudioInputBuffer, float** ppfAudioOutputBuffer, int iNumChannels, int iNumSamples)
```

Comment:
This function is not dry. You are actually copy and pasting exactly the same code in [`displayBuffer`](https://github.com/JohnKasun/2022-MUSI6106/blob/assignment1_combfilter_PeerReview/src/MUSI6106Exec/MUSI6106Exec.cpp#L229), with only the content of `std::cout << "Buffer: " << std::endl;` changed. This creates redundant code and should be avoided. You can add a wrapper function like:

```c++
void displayBufferAndName(float** ppfAudioBuffer, int iNumChannels, int iNumSamples, const std::string& bufferName) {
    std::cout << bufferName << ": " << std::endl;
    displayBuffer(ppfAudioBuffer, iNumChannels, iNumSamples);
}
```

And rewrite `displayIOBuffers` as:

```c++
    void displayIOBuffers(float** ppfAudioInputBuffer, float** ppfAudioOutputBuffer, int iNumChannels, int iNumSamples) {
        displayBufferAndName(ppfAudioInputBuffer, iNumChannels, iNumSamples, "Input");
        displayBufferAndName(ppfAudioOutputBuffer, iNumChannels, iNumSamples, "Output");
    }
```